ops/framework.py

```diff
@@ -1010,6 +1045,70 @@ def __get__(self, parent, parent_type=None):
                 self.__class__.__name__, parent_type.__name__))
 
 
+class ConfigHistoryObserver(Object):
+    """A class used to store config history for current application."""
+    _stored = StoredState()
+
+    def __init__(self, parent):
+        super().__init__(parent, None)
+        self._keys = list(self.model.config.keys())
+        self._depth = -1
+        self._stored.set_default(history={key: [self.model.config[key]] for key in self.keys})
+
+    @property
+    def history(self) -> 'StoredDict':
+        """Config history."""
+        return self._stored.history
+
+    @property
+    def keys(self) -> typing.List[str]:
+        """List of tacked keys."""
+        return self._keys
+
+    @keys.setter
+    def keys(self, keys: typing.List[str]) -> None:
+        """Set list if tacked keys."""
+        _keys = []
+        for key in keys:
+            if key not in self.model.config.keys():
+                raise ValueError("Key {} was not found in application config.".format(key))
+
+            _keys.append(key)
+
+        self._keys = keys
+
+    def set_depth(self, depth: int) -> None:
+        """Set depth of tacked history."""
+        if not isinstance(depth, int) and (depth == -1 or 1 <= depth):
+            raise ValueError("Depth must be integer from range [1, inf). [inf <==> -1]")
```

Comment:
What does zero mean? Only store the current value? Or none at all?
I guess it's forbidden so it's triggered during unit tests instead of showing up as an IndexError when they `self.framework.tack["key-1"][-1]`?

Reply:
The zero will represent storing none at all. So If you try to use `depth == 0` in the `tacked` function, you will end up w/ `ValueError`.